File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if(s1.size() > s2.size()) return false;
        sort(s1.begin(), s1.end());

        for(int i = 0; i < s2.size(); i++) {
            string tmp = "";
            int ok = 0;
            for(int j = i; j < s2.size(); j++) {
                tmp += s2[j];
                if(tmp.size() == s1.size()) {
                    sort(tmp.begin(), tmp.end());
                    if(tmp == s1) {
                        ok = 1;
                        break;
                    }
                }
            }
            if(ok) return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-0.java
class Solution {
    public boolean isAnagram(String s, String t) {
        char[] charS = s.toCharArray();
        char[] charT = t.toCharArray();
        Arrays.sort(charS);
        Arrays.sort(charT);
        s = new String(charS);
        t = new String(charT);
        return s.equals(t);
    }
}
